src/gpio.rs

```diff
@@ -15,6 +18,65 @@ mod gpio_commands {
     pub const DISABLE: usize = 9;
 }
 
+mod subscribe_nr {
+    pub const SUBSCRIBE_CALLBACK: usize = 0;
+}
+
+pub struct GpioDriver {
+    pub(crate) _unconstructible: (),
+}
+
+impl GpioDriver {
+    pub fn all_pins<'a>(&'a mut self) -> TockResult<GpioIter<'a>> {
```

Comment:
It would be great not to change the current API. I prepared a unification of our main APIs and would, therefore, like to avoid modifying the API twice.

Reply:
fied.